File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
       int A = word1.length();
       int B = word2.length();
       int a = 0 , b = 0;
       StringBuilder result = new StringBuilder();// because stringBuilder is mutable

        while(a < A && b < B){
            result.append(word1.charAt(a));
            result.append(word2.charAt(b));
            a++;
            b++;
        }

        while(a < A){
           result.append(word1.charAt(a));
           a++; 
        }

        while(b < B){
           result.append(word2.charAt(b));
           b++;
        }
        return result.toString();// stringbuilder is a different data type;
    }
}

File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
       int word = 1;//for switching the word alterantively
       int A = word1.size();//for traversing the 1st word 
       int B = word2.size();//for traversing the 2nd word
       int a = 0; int b = 0;//pointer for word1 and word2
       string result;// for printing result

       while(a < A && b < B){//while the pointer is not in the end of the word
        result += word1[a];// including the character in the string from the word
        result += word2[b];
        a++; b++;
       }
       while(a < A){// complete the remaining word if left
        result += word1[a];
        a++;
       }
       while(b < B){// complete the remaining word if left
        result += word2[b];
        b++;
       }
    return result;
    }
};